ledger/complete/ledger.go

```diff
@@ -200,20 +262,31 @@ func (l *Ledger) Set(update *ledger.Update) (newState ledger.State, trieUpdate *
 
 	l.metrics.UpdateCount()
 
-	walChan := make(chan error)
+	walChan := make(chan walUpdateResult)
 
 	go func() {
-		walChan <- l.wal.RecordUpdate(trieUpdate)
+		segmentNum, err := l.wal.RecordUpdate(trieUpdate)
```

Comment:
> If two updates are written to the same segment file, then the same segmentNum will be returned again, right?

Yes.

Reply:
Is it possible to return two numbers, the segmentNum and the index of the trie update included in the segment?

So that when subscribing the SegmentTrieUpdate, we can double check the order:
```
SegmentTrie{trie: trie1, segumentNum: 10, indexInSegument: 4}
SegmentTrie{trie: trie3, segumentNum: 10, indexInSegument: 6}
SegmentTrie{trie: trie2, segumentNum: 10, indexInSegument: 5}
```

If we receive the trie updates in the above order, then we know it must be inconsistent with the order in the WAL files.

Reply:
It's possible to include index in PR #2792.  But since PR #2792 handles concurrency and ensures the WAL update and ledger state update are in sync, maybe we don't need to include index to detect inconsistency in #2792.